up_transport_zenoh/zenohutils.py

```diff
@@ -72,77 +70,71 @@ def get_uauth_from_uuri(uri: UUri) -> Union[str, UStatus]:
-            UPriority.UPRIORITY_CS1: Priority.DATA_LOW(),
-            UPriority.UPRIORITY_CS2: Priority.DATA(),
-            UPriority.UPRIORITY_CS3: Priority.DATA_HIGH(),
-            UPriority.UPRIORITY_CS4: Priority.INTERACTIVE_LOW(),
-            UPriority.UPRIORITY_CS5: Priority.INTERACTIVE_HIGH(),
-            UPriority.UPRIORITY_CS6: Priority.REAL_TIME(),
-            UPriority.UPRIORITY_UNSPECIFIED: Priority.DATA_LOW(),
+            UPriority.UPRIORITY_CS0: Priority.BACKGROUND,
+            UPriority.UPRIORITY_CS1: Priority.DATA_LOW,
+            UPriority.UPRIORITY_CS2: Priority.DATA,
+            UPriority.UPRIORITY_CS3: Priority.DATA_HIGH,
+            UPriority.UPRIORITY_CS4: Priority.INTERACTIVE_LOW,
+            UPriority.UPRIORITY_CS5: Priority.INTERACTIVE_HIGH,
+            UPriority.UPRIORITY_CS6: Priority.REAL_TIME,
+            UPriority.UPRIORITY_UNSPECIFIED: Priority.DATA_LOW,
         }
         return mapping[upriority]
 
-    @staticmethod
-    def to_upayload_format(encoding: Encoding) -> UPayloadFormat:
-        try:
-            value = int(encoding.suffix)
-            return value if UPayloadFormat.Name(value) else None
-        except (ValueError, AttributeError):
-            return None
-
     @staticmethod
     def uattributes_to_attachment(uattributes: UAttributes):
-        attachment = [("", UATTRIBUTE_VERSION.to_bytes(1, byteorder='little')), ("", uattributes.SerializeToString())]
-        return attachment
+        # Convert the version number to bytes (assuming 1 as in the Rust example)
+        version_bytes = UATTRIBUTE_VERSION.to_bytes(1, byteorder='little')
+
+        # Serialize the UAttributes to bytes
+        uattributes_bytes = uattributes.SerializeToString()
+
+        # Combine version bytes and uattributes bytes into one list of bytes
+        attachment_bytes = version_bytes + uattributes_bytes
+
+        # Convert the combined bytes to ZBytes
+        return ZBytes(attachment_bytes)
 
     @staticmethod
-    def attachment_to_uattributes(attachment: Attachment) -> UAttributes:
+    def attachment_to_uattributes(attachment: ZBytes) -> UAttributes:
         try:
-            version = None
-            version_found = False
-            uattributes = None
-
-            items = attachment.items()
-            for pair in items:
-                if not version_found:
-                    version = pair[1]
-                    version_found = True
-                else:
-                    # Process UAttributes data
-                    uattributes = UAttributes()
-                    uattributes.ParseFromString(pair[1])
-                    break
-
-            if version is None:
-                msg = f"UAttributes version is empty (should be {UATTRIBUTE_VERSION})"
-                logging.debug(msg)
-                raise UStatusError.from_code_message(code=UCode.INVALID_ARGUMENT, message=msg)
+            # Convert ZBytes to a list of bytes
+            attachment_bytes = bytes(attachment)
```

Comment:
We can use `attachment_lists = attachment.deserialize(list)` to transform the attachment back to the list.